tests/run-make/staticlib-thin-archive/rmake.rs

```diff
@@ -0,0 +1,28 @@
+// Regression test for https://github.com/rust-lang/rust/issues/107407 which
+// checks that rustc can read thin archive. Before the object crate added thin
+// archive support rustc would add emit object files to the staticlib and after
+// the object crate added thin archive support it would previously crash the
+// compiler due to a missing special case for thin archive members.
+use std::path::Path;
+
+use run_make_support::{llvm_ar, rust_lib_name, rustc, static_lib_name};
+
+fn main() {
+    std::fs::create_dir("archive").unwrap();
+
+    // Build a thin archive
+    rustc().input("simple_obj.rs").emit("obj").output("archive/simple_obj.o").run();
+    llvm_ar()
+        .obj_to_thin_ar()
+        .output_input(
+            Path::new("archive").join(static_lib_name("thin_archive")),
```

Comment:
this could be `path("archive")` to avoid the `std::path` import. (add `path` to run_make_support imports)

Reply:
Done